tests/fast-integration/utils/index.js

```diff
@@ -315,6 +315,11 @@ async function k8sApply(resources, namespace, patch = true) {
 }
 
 function waitForK8sObject(path, query, checkFn, timeout, timeoutMsg) {
+  return waitForK8sObjectWithWatch(path, query, checkFn, timeout, timeoutMsg, this.watch)
```

Comment:
I think `this.watch` will be `undefined` if the function is not assigned to any object (global context). Can you prove that the code works at all?
Tests are failing. Linter is complaining.  This PR looks like a draft.

Reply:
The old `waitForK8sObject` seem to operate "semi-global" `this.watch` that is initialised in `initializeK8sClient`. Because of client being initialised on module import in `utils.js` it is hard for me to reinitialised it when it is usually used for provisioned cluster but I need for communication with Gardener. Would you mind me submitting the examples in the following PRs?